spinalcordtoolbox/labels.py

```diff
@@ -0,0 +1,473 @@
+                                               ((z_centerline[indexes_slice[0][index_slice + 1]] - z_centerline[indexes_slice[0][index_slice]]) * pz)**2)
+                                       for index_slice in range(len(indexes_slice[0]) - 1)])
+
+    # 2. for each slice:
+    #   a. identify corresponding vertebral level --> i
+    #   b. calculate distance of slice from upper vertebral level --> d
+    #   c. compute relative distance in the vertebral level coordinate system --> d/Di
+    continuous_values = {}
+    for it, iz in enumerate(z_centerline):
+        level = value_centerline[it]
+        indexes_slice = np.where(value_centerline == level)
+        indexes_slice = indexes_slice[0][indexes_slice[0] >= it]
+        distance_from_level = np.sum([np.sqrt(((x_centerline_fit[indexes_slice[index_slice + 1]] - x_centerline_fit[indexes_slice[index_slice]]) * px * px) ** 2 +
+                                              ((y_centerline_fit[indexes_slice[index_slice + 1]] - y_centerline_fit[indexes_slice[index_slice]]) * py * py) ** 2 +
+                                              ((z_centerline[indexes_slice[index_slice + 1]] - z_centerline[indexes_slice[index_slice]]) * pz * pz) ** 2)
+                                      for index_slice in range(len(indexes_slice) - 1)])
+        continuous_values[iz] = level + 2.0 * distance_from_level / float(length_levels[level])
+
+    # 3. saving data
+    # for each slice, get all non-zero pixels and replace with continuous values
+    coordinates_input = img.getNonZeroCoordinates()
+    out.change_type(np.float32)
+
+    # for all points in input, find the value that has to be set up, depending on the vertebral level
+    for x, y, z, v in coordinates_input:
+        out.data[int(x), int(y), int(z)] = continuous_values[z]
+
+    if out.orientation != og_orientation:
+        out.change_orientation(og_orientation)
+        img.change_orientation(og_orientation)
+
+    return out
+
+
+def remove_labels_from_image(img: Image, labels: Sequence[int]) -> Image:
+    """
+    Remove specified labels (set to 0) from an image.
+    :param img: source image
+    :param labels: list of specified labels to remove
+    :returns: image with labels specified removed
+    """
+    out = img.copy()
+
+    for l in labels:
+        for x, y, z, v in img.getNonZeroCoordinates():
+            if l == v:
+                out.data[int(x), int(y), int(z)] = 0.0
+                break
+        else:
+            logger.warning(f"Label {l} not found in input image!")
+
+    return out
+
+
+def remove_other_labels_from_image(img: Image, labels: Sequence[int]) -> Image:
+    """
+    Remove labels other than specified from an image
+    :param img: source image
+    :param labels: list of specified labels to keep
+    :returns: image with labels specified kept only
+    """
```

Comment:
Could this function just be a parameter in `remove_labels_from_image`?

For example `inverse` with description `If inverse=True, this removes all labels *except* the ones provided`.

Reply:
It could, and this may be just a question of style, but I like to have functions as specific as possible (SRP) in order to simplify tests and maintenance. 

I though that since we are iterating over potentially _a lot_ of labels (depending on size on input image) the check for `inverse` would unnecessarily slow down the code.

Reply:
That seems very fair to me. And you're right, it is mostly a nitpicky style thing, so I don't think it's a big deal. :)